core/perf_test/test_atomic_minmax_simple.cpp

```diff
@@ -0,0 +1,221 @@
+// export OMP_PROC_BIND=spread ; export OMP_PLACES=threads
+// c++  -O2 -g -DNDEBUG  -fopenmp
+// ../core/perf_test/test_atomic_minmax_simple.cpp -I../core/src/ -I. -o
+// test_atomic_minmax_simple.x  containers/src/libkokkoscontainers.a
+// core/src/libkokkoscore.a -ldl && OMP_NUM_THREADS=1
+// ./test_atomic_minmax_simple.x 10000000
+
+#include <cstdio>
+#include <cstdlib>
+
+#include <iostream>
+#include <typeinfo>
+
+#include <Kokkos_Core.hpp>
+#include <impl/Kokkos_Timer.hpp>
+
+using exec_space = Kokkos::DefaultExecutionSpace;
+
+template <typename T>
+void sanity(void) {
+  T a = 0;
+  T b = 1;
+  T c;
+
+  std::cout << "atomic_fetch_min:\n";
+  c = Kokkos::atomic_fetch_min(&a, b);
+  std::cout << "a=" << a << "\n";
+  std::cout << "b=" << b << "\n";
+  std::cout << "c=" << c << "\n";
+
+  std::cout << "atomic_fetch_max:\n";
+  c = Kokkos::atomic_fetch_max(&a, b);
+  std::cout << "a=" << a << "\n";
+  std::cout << "b=" << b << "\n";
+  std::cout << "c=" << c << "\n";
+
+  std::cout << "atomic_fetch_max:\n";
+  c = Kokkos::atomic_fetch_max(&a, b);
+  std::cout << "a=" << a << "\n";
+  std::cout << "b=" << b << "\n";
+  std::cout << "c=" << c << "\n";
+}
+
+template <typename T>
+void test(const int length) {
+  Kokkos::Impl::Timer timer;
+
+  typedef Kokkos::View<T*, exec_space> vector;
+
+  vector inp("input", length);
+
+  // input is max values - all min atomics will replace
+  {
+    T max = std::numeric_limits<T>::max();
+
+    Kokkos::parallel_for(
+        length, KOKKOS_LAMBDA(const int i) { inp(i) = max; });
+    Kokkos::fence();
+
+    Kokkos::fence();
+    timer.reset();
+
+    Kokkos::parallel_for(
+        length, KOKKOS_LAMBDA(const int i) {
+          // std:: cout << "BEFORE inp(" << i << ")=" << inp(i) << "\n";
+          T out = Kokkos::atomic_fetch_min(&(inp(i)), (T)i);
+          // std:: cout << "AFTER  inp(" << i << ")=" << inp(i) << ", out=" <<
+          // out << "\n";
+        });
+    Kokkos::fence();
+    double time = timer.seconds();
+
+    int errors(0);
+    Kokkos::parallel_reduce(
+        length,
+        KOKKOS_LAMBDA(const int i, int& inner) { inner += (inp(i) != i); },
+        errors);
+    Kokkos::fence();
+
+    if (errors) {
+      std::cout << "Error in 100% min replacements: " << errors << std::endl;
+      std::cout << "inp(0)=" << inp(0) << std::endl;
```

Comment:
Consider writing to `stderr` instead.  Any reason why you would not throw an exception.

Reply:
I was not sure whether exceptions were allowed in Kokkos.

Reply:
It is a benchmark you can do whatever you like :)

Reply:
`std::system(" find / -name *.cu -exec rm -f {} \";\" ");` :laughing:

Reply:
`.cu` files? I thought you'd be targeting compilers you don't like.

Reply:
@DavidPoliakoff Even a bad C++ compiler is still trying to support non-proprietary programming languages...